modules/core/include/opencv2/core/hal/intrin_math.hpp

```diff
@@ -34,434 +30,658 @@ namespace CV__SIMD_NAMESPACE {
-        const v_float64 _vexp_q0_f64 = vx_setall_f64(3.00198505138664455042E-6);
-        const v_float64 _vexp_q1_f64 = vx_setall_f64(2.52448340349684104192E-3);
-        const v_float64 _vexp_q2_f64 = vx_setall_f64(2.27265548208155028766E-1);
-        const v_float64 _vexp_q3_f64 = vx_setall_f64(2.00000000000000000009E0);
-        const v_int64 _vexp_bias_s64 = vx_setall_s64(0x3ff);
-
-        v_float64 _vexp_, _vexp_x, _vexp_y, _vexp_z, _vexp_xx;
-        v_int64 _vexp_mm;
-
-        // compute exponential of x
-        _vexp_x = v_max(x, _vexp_lo_f64);
-        _vexp_x = v_min(_vexp_x, _vexp_hi_f64);
-
-        _vexp_ = v_fma(_vexp_x, _vexp_LOG2EF_f64, _vexp_half_f64);
-        _vexp_mm = v_expand_low(v_floor(_vexp_));
-        _vexp_ = v_cvt_f64(_vexp_mm);
-        _vexp_mm = v_add(_vexp_mm, _vexp_bias_s64);
-        _vexp_mm = v_shl(_vexp_mm, 52);
-
-        _vexp_x = v_fma(_vexp_, _vexp_C1_f64, _vexp_x);
-        _vexp_x = v_fma(_vexp_, _vexp_C2_f64, _vexp_x);
-        _vexp_xx = v_mul(_vexp_x, _vexp_x);
-
-        _vexp_y = v_fma(_vexp_xx, _vexp_p0_f64, _vexp_p1_f64);
-        _vexp_y = v_fma(_vexp_y, _vexp_xx, _vexp_p2_f64);
-        _vexp_y = v_mul(_vexp_y, _vexp_x);
-
-        _vexp_z = v_fma(_vexp_xx, _vexp_q0_f64, _vexp_q1_f64);
-        _vexp_z = v_fma(_vexp_xx, _vexp_z, _vexp_q2_f64);
-        _vexp_z = v_fma(_vexp_xx, _vexp_z, _vexp_q3_f64);
-
-        _vexp_z = v_div(_vexp_y, v_sub(_vexp_z, _vexp_y));
-        _vexp_z = v_fma(_vexp_two_f64, _vexp_z, _vexp_one_f64);
-        _vexp_z = v_mul(_vexp_z, v_reinterpret_as_f64(_vexp_mm));
-
-        // exp(NAN) -> NAN
-        v_float64 mask_not_nan = v_not_nan(x);
-        return v_select(mask_not_nan, _vexp_z, v_reinterpret_as_f64(vx_setall_s64(0x7FF8000000000000)));
-    }
-#endif
-
-#define OPENCV_HAL_MATH_HAVE_EXP 1
-//! @}
-#endif
+// Implementation is the same as float32 vector.
+template<typename _TpVec16F, typename _TpVec16S>
+inline _TpVec16F v_exp_default_16f(const _TpVec16F &x) {
+    const _TpVec16F _vexp_lo_f16 = v_setall_<_TpVec16F>(-10.7421875f);
+    const _TpVec16F _vexp_hi_f16 = v_setall_<_TpVec16F>(11.f);
+    const _TpVec16F _vexp_half_fp16 = v_setall_<_TpVec16F>(0.5f);
+    const _TpVec16F _vexp_one_fp16 = v_setall_<_TpVec16F>(1.f);
+    const _TpVec16F _vexp_LOG2EF_f16 = v_setall_<_TpVec16F>(1.44269504088896341f);
+    const _TpVec16F _vexp_C1_f16 = v_setall_<_TpVec16F>(-6.93359375E-1f);
+    const _TpVec16F _vexp_C2_f16 = v_setall_<_TpVec16F>(2.12194440E-4f);
+    const _TpVec16F _vexp_p0_f16 = v_setall_<_TpVec16F>(1.9875691500E-4f);
+    const _TpVec16F _vexp_p1_f16 = v_setall_<_TpVec16F>(1.3981999507E-3f);
+    const _TpVec16F _vexp_p2_f16 = v_setall_<_TpVec16F>(8.3334519073E-3f);
+    const _TpVec16F _vexp_p3_f16 = v_setall_<_TpVec16F>(4.1665795894E-2f);
+    const _TpVec16F _vexp_p4_f16 = v_setall_<_TpVec16F>(1.6666665459E-1f);
+    const _TpVec16F _vexp_p5_f16 = v_setall_<_TpVec16F>(5.0000001201E-1f);
```

Comment:
use `hfloat(xxxx.f)` to replace the input number
`v_setall_<_TpVec16F>(hfloat(-6.93359375E-1f));`